pymeasure/instruments/newport/smc100.py

```diff
@@ -0,0 +1,707 @@
+        message = "Newport SMC100 Motion Controller with the following\n"
+        message += "settings (u stands for mm or deg by default):\n"
+        message += f"* address: {self.address}\n"
+        for k, v in self.__class__.__dict__.items():
+            if type(v) is property and v.fget.__defaults__[0] is not None:
+                unit = ""
+                try:
+                    unit += v.__doc__.split("Unit:")[1].split("\n")[0]
+                except IndexError:
+                    pass
+                message += f"* {k}: {v.fget(self)}{unit}\n"
+        return message[:-1]
+
+    # Implementation or extension of Instrument parent class methods
+    def check_errors(self):
+        """Check for command error except when broadcasting."""
+        if self.address != "0":
+            self.check_command_error()
+
+    def ask(self, command):
+        return super().ask(self.address + command)
+
+    def shutdown(self):
+        """Stop any motion and put the controller in DISABLE state."""
+        self.stop()
+        self.wait_ready()
+        self.disable = True
+        super().shutdown()
+
+    def values(self, command, **kwargs):
+        return super().values(self.address + command, **kwargs)
+
+    def write(self, command):
+        super().write(self.address + command)
+
+    # Other methods
+    def check_command_error(self):
+        """Check for and raise any command error."""
+        error_code = self.error_command
+        try:
+            assert error_code == '@'
+        except AssertionError:
+            raise(CommandError(self.address, error_code))
+
+    def check_motion_error(self):
+        """Check for and raise any motion error.
+
+        It implements this by verifying that the controller doesn't
+        unexpectedly end up in DISABLE or NOT REFERENCED states, which
+        indicates that an error happened during homing or moving. Using
+        this command may hide positioner errors.
+        """
+        state = self.state
+        try:
+            assert not state.disable and not state.not_referenced
+        except AssertionError:
+            raise(MotionError(self.address, state))
+
+    def check_positioner_error(self):
+        """Check for and raise any positioner error."""
+        errors = self.error_positioner
+        try:
+            assert not any(errors)
+        except AssertionError:
+            raise(PositionerError(self.address, errors))
+    
+    def get_motion_time(self, distance):
+        """Get an evaluation of the time needed to complete a
+        relative move.
+
+        Argument: distance of the move in predefined unit.
+        """
+        return self.values("PT" + str(abs(distance)))[0]
+
+    def reset(self):
+        """Reset the controller."""
+        self.write("RS")
+        sleep(1)
+        # Sequence necessary to avoid error on next command
+        try:
+            self.check_errors()
+        except VisaIOError:
+            pass
+
```

Comment:
Increasing the sleep time, it does not help to avoid the error on next command right ?
I think, even not ideal, it could expected that after reset the instrument RS232 interface is not responsive, but I would expect that become responsive after a while, so I could expect that enlarging the `sleep` could help.
On the other hand, if enlarging the `sleep` time does not work, I wonder if the sleep is needed at all.
Do you know what kind of `VisaIOError` is throwing ?

Reply:
Sorry for the late reply. I tried to remove either the sleep time, or the error query (with increased sleep time), and neither work. In fact, after a lot of repetitions, the sleep time seamed insufficient in some cases, so I increased it.